aws/resource_aws_instance.go

```diff
@@ -893,6 +893,10 @@ func resourceAwsInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
 				groups = append(groups, aws.String(v.(string)))
 			}
 		}
+
+		if len(groups) < 1 {
```

Comment:
Minor nitpick: In a future enhancement, we should be able to move this into a `CustomizeDiff` function to perform this validation during planning 👍